tools/check.py

```diff
@@ -144,39 +143,21 @@
 
 
 def add_error(msg, errors):
-    """
-    Add error to the list of errors.  This function makes use of Python
-    reference mechanics: a list passed as `errors` argument is always passed
-    as a reference, therefore modification of that list in this function will
-    result in the same modification in the calling (parent) function.
-
-    For example:
-    >>> L = []
-    >>> add_error("Test1", L)
-    >>> print(L)
-    ['Test1']
-    """
+    """Add error to the list of errors."""
     errors.append(msg)
 
 
 def add_suberror(msg, errors):
-    """
-    Add sub error, ie. error indented by 1 level ("\t"), to the list of errors.
-    To see how this function works, take a look at :ref:`add_error` function.
-
-    An example:
-    >>> print(L)
-    ['Test1']
-    >>> add_suberror("Test1_1", L)
-    >>> print(L)
-    ['Test1', '\tTest1_1']
-    """
+    """Add sub error, ie. error indented by 1 level ("\t"), to the list of errors."""
     errors.append("\t{0}".format(msg))
 
+
 def look_for_fixme(func):
-    '''Decorator to see whether a value starts with FIXME.'''
+    '''Decorator to fail test if text argument starts with "FIXME".'''
     def inner(arg):
-        if (arg is not None) and arg.lstrip().startswith('FIXME'):
+        if (arg is not None) and \
+           isinstance(arg, basestring) and \
```

Comment:
No problem with Python 2.7.8 but

~~~
$ python --version
Python 3.4.2
$ python check.py 
INFO: Testing "../index.html"
Traceback (most recent call last):
  File "check.py", line 457, in <module>
    main()
  File "check.py", line 445, in main
    errors = check_file(filename, data)
  File "check.py", line 408, in check_file
    error_message)
  File "check.py", line 337, in check_validity
    valid = function(data)
  File "check.py", line 159, in inner
    isinstance(arg, basestring) and \
NameError: name 'basestring' is not defined
~~~

Replace `basestring` with `str` works for Python 2 and 3.